NuRadioReco/modules/channelGenericNoiseAdder.py

```diff
@@ -130,18 +134,24 @@ def bandlimited_noise(self, min_freq, max_freq, n_samples, sampling_rate, amplit
 #         f1 = plt.figure()
 #         plt.plot (frequencies/max(frequencies))
 #         plt.plot(fbinsactive,'kx')
-
-        if(bandwidth is not None):
+        ampl = np.zeros(n_samples_freq)
+        if callable(amplitude):
+            ampl[selection] = amplitude(frequencies[selection])
+        elif (bandwidth is not None):
             sampling_bandwidth = min(0.5 * sampling_rate, max_freq) - min_freq
-            amplitude *= 1. / (bandwidth / (sampling_bandwidth)) ** 0.5  # normalize noise level to the bandwidth its generated for
+            ampl[selection] = amplitude / (bandwidth / (sampling_bandwidth)) ** 0.5  # normalize noise level to the bandwidth its generated for
+        else:
+            ampl[selection] = amplitude
 
-        ampl = np.zeros(n_samples_freq)
-        sigscale = (1. * n_samples) / np.sqrt(nbinsactive)
+        if callable(amplitude):
+            sigscale = sampling_rate
+        else:    
+            sigscale = (1. * n_samples) / np.sqrt(nbinsactive)
```

Comment:
These four lines can maybe be included in the if statements above (line 137 to 144) as this is just another scaling. It should maybe also be applied there.